src/lib.rs

```diff
@@ -84,15 +101,16 @@ impl<'a, T: Clone> CBuf<'a, T>
     ///
     /// Returns `None` if buffer is empty.
     #[inline]
-    pub fn peek(&mut self) -> Option<T> {
+    pub fn peek(&mut self) -> Option<&T> {
         self.ctrl.peek(self.buf)
     }
 
     /// Peek next element from the CBuf without removing it
     ///
-    /// Makes the buffer misbehave if it's empty.
+    /// if the buffer is empty, undefined data will be
+    /// returned.
     #[inline]
-    pub fn peek_unchecked(&mut self) -> T {
+    pub fn peek_unchecked(&mut self) -> &T {
```

Comment:
I added docs to it -- it is no more unsafe than having access to a raw pointer (which is not unsafe). The data it returns may be garbage... but that in itself isn't unsafe either.

On the other hand, garbage data could have unintended side-effects, so maybe it should be unsafe. I'll change it now.